### PR TITLE
ARROW-17435: [CI][Python][CUDA] Install Numba for CUDA interop tests 

### PR DESCRIPTION
**Pull request by pitrou**
Checked locally on a Ubuntu 20.04 host with:
```
archery docker run ubuntu-cuda-python
```

### CODE REVIEW

**Comment by pitrou** [comment]
This is based on PR #13896.

cc @assignUser @kszucs

File: ci/docker/linux-apt-python-3.dockerfile [hunk 1]
```diff
@@ -39,6 +39,12 @@ RUN pip install \
     -r arrow/python/requirements-build.txt \
     -r arrow/python/requirements-test.txt
 
+ARG numba
+COPY ci/scripts/install_numba.sh /arrow/ci/scripts/
+RUN if [ "${numba}" != "" ]; then \
+        /arrow/ci/scripts/install_numba.sh ${numba} \
+    ; fi
```
Comment (pitrou) [inline ci/docker/linux-apt-python-3.dockerfile:46]: Not sure this is ideal, but It wasn't able to create a child Dockerfile with this file as BASE (I may be misunderstanding how these images are managed).

Comment (assignUser) [reply ci/docker/linux-apt-python-3.dockerfile:46]: A good example on child images/services is the `ubuntu-r` one, which builds on the `ubuntu-cpp` dockerfile in it's own `linux-apt-r.dockerfile` 
https://github.com/apache/arrow/blob/master/docker-compose.yml#L1234-L1242

Comment (pitrou) [reply ci/docker/linux-apt-python-3.dockerfile:46]: Hmm... can you suggest a patch? I already tried to follow this pattern and didn't succeed. I got this error:
```
DEBUG:archery:Executing `['docker-compose', '--file', '/home/antoine/arrow/dev/docker-compose.yml', 'build', '--build-arg', 'BUILDKIT_INLINE_CACHE=1', 'ubuntu-cuda-python']`
Building ubuntu-cuda-python
[+] Building 0.4s (3/3) FINISHED                                                                                                                                     
 => [internal] load build definition from linux-apt-python-numba.dockerfile                                                                                     0.0s
 => => transferring dockerfile: 1.04kB                                                                                                                          0.0s
 => [internal] load .dockerignore                                                                                                                               0.0s
 => => transferring context: 35B                                                                                                                                0.0s
 => ERROR [internal] load metadata for docker.io/apache/arrow-dev:amd64-linux-apt-python-3                                                                      0.4s
------
 > [internal] load metadata for docker.io/apache/arrow-dev:amd64-linux-apt-python-3:
------
failed to solve with frontend dockerfile.v0: failed to create LLB definition: docker.io/apache/arrow-dev:amd64-linux-apt-python-3: not found
ERROR: Service 'ubuntu-cuda-python' failed to build : Build failed
Error: `docker-compose --file /home/antoine/arrow/dev/docker-compose.yml build --build-arg BUILDKIT_INLINE_CACHE=1 ubuntu-cuda-python` exited with a non-zero exit code 1, see the process log above.

```

Comment (pitrou) [reply ci/docker/linux-apt-python-3.dockerfile:46]: (side note: it's not help ful that browsing to `docker.io/apache/arrow-dev` does not let me view the available images/tags)

**Comment by kszucs** [comment]
Can you post a link from the logs showing that the numba tests are running?

**Review by kszucs — APPROVED** [review]

**Comment by pitrou** [comment]
> Can you post a link from the logs showing that the numba tests are running?

Here is the output of `archery --debug docker run ubuntu-cuda-python` here:
https://gist.github.com/pitrou/53ae68d6720fe790ca648de05974e7e5

**Comment by kszucs** [comment]
> > Can you post a link from the logs showing that the numba tests are running?
> 
> Here is the output of `archery --debug docker run ubuntu-cuda-python` here: https://gist.github.com/pitrou/53ae68d6720fe790ca648de05974e7e5

Thanks! Feel free to merge it on green.

**Pull request merged**